language/documentation/book/src/address.md

```diff
@@ -6,17 +6,60 @@ Although an `address` is a 128 bit integer under the hood, Move addresses are in
 
 You can use runtime address values (values of type `address`) to access resources at that address. You *cannot* access modules at runtime via address values.
 
-## Literals
+## Addresses and Their Syntax
 
-`address` literals are 16-byte hex literals, i.e. `0x<hex encoded value>`. For convenience, leading `0`s are added for literals that are too short.
+Addresses come in two flavors, named or numerical. The syntax for a named address follows the
+same rules for any named identifier in Move. The syntax of a numerical address is not restricted
+to hex-encoded values, and any valid [`u128` numerical value](./integers.md) can be used as an
+address value, e.g., `42`, `0xCAFE`, and `2021` are all valid numerical address
+literals.
+
+To distinguish when an address is being used in an expression context or not, the
+syntax when using an address will differ slightly depending on the context it's
+being used in:
+* When an address is used as an expression the address must be prefixed by the `@` character, i.e., `@`[`<numeric_value>`](./integers.md)` or `@<named_address_identifier>`.
+* Outside of expression contexts, the address may be written without the leading `@` character, i.e., [`<numeric_value>`](./integers.md) or `<named_address_identifier>`.
+
+In general, you can think of `@` as an operator that takes an address from being a namespace item to being an expression item.
+
+## Named Addresses
+
+As mentioned above, Move supports the concept of named addresses. Named
+addresses are a feature that allow identifiers to be used in place of numerical
+values in any spot where addresses are used, and not just at the value level.
+Named addresses are declared and bound as top level elements (outside of modules and
+scripts) in Move Packages, or passed as arguments to the Move compiler.
+
+Named addresses only exist at the source language level and during compilation
+and will be fully substituted for their value at the bytecode level. Because of
+this, their value should be treated opaquely if at all possible -- and the
```

Comment:
I have multiple issues with this "Because...." sentence (e.g., "value" should be plural, when it is not possible?) I think it would be best to just skip this and cut to the chase in the following sentence:

Because of this, modules and module members _must_ be accessed....